main.rb

```diff
@@ -0,0 +1,147 @@
+# rubocop: disable Metrics/ModuleLength:
+# rubocop: disable Style/For:
+# rubocop: disable Style/CaseEquality:
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    item = 0
+    array = to_a
+    n = array.length
+    while item < n
+      yield(array[item])
+      item += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+    array
```

Comment:
* [ ] should have working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- `[bug location]` Where is the `array` you are returning coming from?
- Main issue:
```ruby
array = Array.new(100) { rand(0...100) }
my_each_output = ''
block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
array.each_with_index(&block)
each_output = my_each_output.dup
my_each_output = ''
array.my_each_with_index(&block)

each_output != my_each_output # they should be equal
```